File: Algorithms/cpp/338 - Counting Bits.cpp
class Solution {
public:
    vector<int> countBits(int num) {            
        // var
        vector<int> result({ 0, 1, 1 });
        
        if (num < 3)
            return vector<int>(result.begin(), result.begin() + num + 1);
        
        for (int i = 3; i <= num; i++)
            result.push_back(result[i >> 1] + (i & 1));
        
        return result;
    }
};

File: python/338 - Counting Bits.py
class Solution(object):
    def countBits(self, num):
        """
        :type num: int
        :rtype: List[int]
        """
        ans = []
        for i in range(0, num + 1):
            # decimal to binary
            bits = list(bin(i)[2:])
            
            count = 0
            for x in bits:
                if x == '1':
                    count += 1
            ans.append(count)
        return ans